### PR TITLE
add xdp test

### PR DESCRIPTION
**Pull request by yanchaomei**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: bpf/kmesh/workload/xdp.c [hunk 1]
```diff
@@ -12,165 +12,123 @@
 #include "config.h"
 #include "bpf_log.h"
 #include "workload.h"
-#include "authz.h"
-#include "xdp.h"
+
```
Comment (bitcoffeeiux) [inline bpf/kmesh/workload/xdp.c:15]: Why would the original program be modified?

File: bpf/kmesh/workload/xdp.c [hunk 2]
```diff
@@ -179,25 +137,16 @@ int xdp_shutdown(struct xdp_md *ctx)
     struct xdp_info info = {0};
     struct bpf_sock_tuple tuple_info = {0};
 
-    if (parser_xdp_info(ctx, &info) == PARSER_FAILED)
+    // 使用全局变量 g_ctx 替代 ctx，方便测试
+    if (parser_xdp_info(&g_ctx, &info) == PARSER_FAILED)
```
Comment (bitcoffeeiux) [inline bpf/kmesh/workload/xdp.c:141]: Normal functions will be affected.

File: bpf/test/workload/xdp_test.c [hunk 3]
```diff
@@ -0,0 +1,114 @@
+// test_xdp.c
```
Comment (bitcoffeeiux) [inline bpf/test/workload/xdp_test.c:1]: Coverage is missing.

**Comment by yanchaomei** [comment]
测试效果：
由于最新的xdp引入了auth，暂时还未适配，目前适配版本为九月份的xdp版本
image
image

**Pull request merged**